File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class ActivityPoint(BaseModel):
    date: str
    questions_asked: int


class MostActivePaper(BaseModel):
    paper_id: int
    title: str
    questions_asked: int


class MostCitedChunk(BaseModel):
    paper_id: int
    paper_title: str
    page_start: int
    page_end: int
    times_cited: int


class AnalyticsSummaryResponse(BaseModel):
    total_papers: int
    total_questions_asked: int
    most_active_paper: MostActivePaper | None
    most_cited_chunks: list[MostCitedChunk]
    activity_by_day: list[ActivityPoint]

File: frontend/src/lib/analytics.ts
export type ActivityPoint = {
  date: string;
  questions_asked: number;
};

export type MostActivePaper = {
  paper_id: number;
  title: string;
  questions_asked: number;
};

export type MostCitedChunk = {
  paper_id: number;
  paper_title: string;
  page_start: number;
  page_end: number;
  times_cited: number;
};

export type AnalyticsSummaryResponse = {
  total_papers: number;
  total_questions_asked: number;
  most_active_paper: MostActivePaper | null;
  most_cited_chunks: MostCitedChunk[];
  activity_by_day: ActivityPoint[];
};

const apiBaseUrl = import.meta.env.VITE_API_BASE_URL ?? 'http://localhost:8000/api/v1';

export async function getAnalyticsSummaryApi(token: string): Promise<AnalyticsSummaryResponse> {
  const response = await fetch(`${apiBaseUrl}/analytics/summary`, {
    headers: { Authorization: `Bearer ${token}` },
  });

  if (!response.ok) {
    const errorBody = (await response.json().catch(() => null)) as { detail?: string } | null;
    throw new Error(errorBody?.detail ?? `Request failed with status ${response.status}`);
  }

  return response.json() as Promise<AnalyticsSummaryResponse>;
}
